packages/console/src/components/SpacesList.tsx

```diff
@@ -37,24 +46,60 @@ interface SpaceItemProps {
 }
 
 function SpaceItem({ space, type }: SpaceItemProps) {
+  const TypeIcon = type === 'private' ? LockClosedIcon : GlobeAltIcon
+  const did = space.did()
+  const truncatedDid = truncateDid(did)
+  
+  const handleCopyDid = async (e: React.MouseEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    try {
+      await navigator.clipboard.writeText(did)
```

Comment:
yea we're doing this in multiple places now - I'd really like to DRY this out and make the copy UX consistent